src/core/weights.js

```diff
@@ -48,19 +52,58 @@ export function copy(w: Weights): Weights {
   };
 }
 
-/**
- * Merge multiple Weights together.
+/** Merge multiple Weights together.
+ *
+ * The resultant Weights will have every weight specified by each of the input
+ * weights.
  *
- * The resultant Weights will have every weight specified by each of the
- * input weights. If there are any overlaps (i.e. the same address is present
- * in two or more of the input weights), an error will be thrown. In the future,
- * we will likely modify this function to add a resolver that determines how to
- * combine multiple overlapping weights.
+ * When there are overlaps (i.e. the same address is present in two or more
+ * input weights), the appropriate resolver will be invoked to resolve the
+ * conflict. The resolver will receive, as its first argument, the result of
+ * resolving the last conflict (or the first weight if there have been no
+ * conflicts), and its second argument will be the latest conflicting weight.
+ * Whatever it returns will be used as the weight (modulo future conflicts).
```

Comment:
The process is very similar to a [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) function.
Perhaps the terms make sense to use as well. Accumulator and current value.

Adding these semantics would be nice as comments on the NodeOperator and EdgeOperator types as well.

Reply:
Sure, I can update the docs to use this vocabulary; PTAL.

Regarding adding comments directly to the types-- the merge function uses them as reducers, but the types themselves could be used for other purposes too. So I don't think we should add comments directly at the type level.

Reply:
I think that “reducers” in the JavaScript sense of the word give the
wrong impression, precisely because “accumulator and current value” is
the wrong formulation. The reduction operator should be associative, and
should be able to combine values in any order: `f(f(x, y), z)` should be
the same as `f(x, f(y, z))`. (Note that I’m not saying that it needs to
be commutative.)

(I recommend watching [“Four Solutions to a Trivial Problem”][4s] if you
haven’t seen it, but parallelism isn’t the only goal. Reductions in the
OCaml sense are far simpler to think about than “folds”.)

[4s]: https://www.youtube.com/watch?v=ftcIcn8AmSY

Reply:
Interesting talk, thanks :smile: 
I did notice this awkwardness with javascript's "reduce function" signature. Given JS's history, can picture why going for a fold instead of reduce makes sense.

Anyway, I didn't consider suggesting an associative function. The docstring explicitly described an accumulator and current value, just in different terms. Agreed that in this case, we may not need these concepts and could require the function to be associative instead.

Reply:
Marking this as resolved as the terms are more clear now.
Will defer to @wchargin whether it should be associative instead.